tools/rna_tools/cleaverna/cleaverna.xml

```diff
@@ -0,0 +1,636 @@
+                        </param>
+                        <!-- Uses bundled SARS_param.csv and FASTA files: 1.fasta through 20.fasta -->
+                        <!-- Default parameters: LA=16, RA=8, CS="AU,GU", Tem=23, CA="GGCUAGCUACAACGA" -->
+                        <param name="sars_info" type="hidden" value="SARS model uses default parameters: LA=16, RA=8, CS=AU,GU, Tem=23, CA=GGCUAGCUACAACGA"/>
+                        <param name="LA" type="text" value="16" label="Left Arm (LA) - Default" help="Fixed default value for SARS model (cannot be changed)"/>
+                        <param name="RA" type="text" value="8" label="Right Arm (RA) - Default" help="Fixed default value for SARS model (cannot be changed)"/>
+                        <param name="CS" type="text" value="AU,GU" label="Cleavage Sites (CS) - Default" help="Fixed default value for SARS model (cannot be changed)"/>
+                        <param name="Tem" type="text" value="23" label="Temperature (Tem) - Default" help="Fixed default value for SARS model (cannot be changed)"/>
+                        <param name="CA" type="text" value="GGCUAGCUACAACGA" label="Catalytic Core (CA) - Default" help="Fixed default value for SARS model (cannot be changed)"/>
+                    </when>
+                    <when value="custom_files">
+                           <param argument="model_name" type="text" value="" label="Model Name" 
+                               help="Enter a name for your model. Suggestion: use a descriptive name (e.g. MyCustomModel)">
+                            <validator type="empty_field" message="Model name is required"/>
+                            <validator type="regex" message="Model name must contain only letters, numbers, and underscores">^[a-zA-Z0-9_]+$</validator>
+                           </param>
+                           <param argument="training_target_files" type="data" format="fasta" multiple="true"
+                               label="Training RNA sequence files" 
+                               help="Specify one or more RNA sequence files in FASTA format for training."/>
+                        <conditional name="params_source">
+                            <param name="input_type" type="select" label="Parameter input method">
+                                <option value="individual" selected="true">Enter individual parameters</option>
+                                <option value="file">Upload CSV file</option>
+                            </param>
+                            <when value="individual">
+                                <param name="LA" type="integer" min="1" max="50" value="15" label="LA - Left arm length" help="The length of the left binding arm of DNAzyme (1-50 nucleotides)">
+                                    <validator type="in_range" min="1" max="50" message="Left arm length must be between 1 and 50"/>
```

Comment:
I think you can remove the validator since you restirct the main param with min/max